### DISCUSSION TITLE
How to crop text if it's over a given size using +annotate command?

### DISCUSSION THREAD

**Discussion by SolarGuy8** [ImageMagick/ImageMagick#5904] [Help]
### ImageMagick version

7.1.0

### Operating system, version and so on

MacOS 13.1

### Description

Hi! My command:
`convert a.jpg \
-font "Arial-Black" -pointsize 75 -fill white -gravity northwest -annotate +1135+1060 "3 Tastenice Tastenice Tastenice" \
-font "Arial-Black" -pointsize 75 -fill white -gravity northwest -annotate +1135+1220 "4 Happy New Year!!!" \
b.jpg`

I have a problem, that text in `+annotate` is dynamic and I want to crop text (or add three dots at the end) if text is exceed 20 characters. How can I do this?

**Comment by snibgo** [comment]
You would need to do that in a shell script. First, detect whether the text at the given pointsize would fit in an image of that size. That can be done by making a white image, annotate and trim. Repeat but with a white image of twice the dimensions. If the results are identical, then the text did fit.

From that, your script might amend the text, shortening it and adding three dots.
